static_default_ports.go

```diff
@@ -22,7 +22,8 @@ func AddressWithDefault(addr, defaultPort string) string {
 	// off first
 	parts := strings.SplitN(addr, "{", 2)
 	endpointPart := parts[0]
-	var templatePart string
+	// We set it as a string because there is no else and if no else then undefined without set
+	var templatePart string = ""
```

Comment:
You don't really need the `= ""` here, it's already the default value of a string

Reply:
Interesting you say that because I ran it without that and with and it was technically undefined as it wasn't initialized at first and it said undefined. I figured undefined was worse than setting it as string and null are not the same in go. My IDE is pointing it out as well

Reply:
Your IDE is wrong :D The two lines are identical as the zero value for a string really is the empty string. It doesn't need to be intialized. See https://go.dev/tour/basics/12

Reply:
Haha. Sounds good do you want me to remove it?

Reply:
Yes please. It doesn't really do anything bad, but this kind of initialization is typically discouraged in Go

Reply:
Updated